en/developer_guide/bbcode.rst

```diff
@@ -0,0 +1,153 @@
+BBCode parser.
+
+It's important to note that by default, not all tags are enabled in BBCode format, only those you
+want to declare. For instance, putting a text in bold in the rich text editor does not produce by
+default a [b] tag. Out of the box, there is only one BBCode: ``media``. But you can add your own.
+
+In this documentation you will see how the BBCode feature fits in the main processes and how to create
+a new BBCode in your bundle.
+
+Wysiwyg process
+---------------
+Open Orchestra uses TinyMce to render a rich text input. Normally, TinyMce generates html tags, but
+Open Orchestra has enhanced it to use BBCode. Here is the workflow triggered when editing an object
+using a form containing rich text fields:
+
+.. image:: ../../images/richtext2.png
+
+1. Each rich text field goes into a data transformer to interpret the BBCode tags. This data
+   transformer calls the Open Orchestra BBCode parser to detect and transform valid BBCode tags into
+   html tags.
+2. The resulting html is given to the TinyMce editor who treat it with its standard logic.
+3. When the user clicks in the toolbar to inject a BBCode tag (for instance by selecting a media),
+   TinyMce inject the html representation of the BBCode, to be able to visually render the tag. For
+   instance in the media case, you will see it in the editor.
+4. When the user saves the form, TinyMce replaces via javascript the html tags by their BBCode
+   representation.
+
+How to create a new BBCode and add it to the parser
+---------------------------------------------------
+Open Orchestra provides a simple mechanism to add your own custom BBCode tags. In fact there is two
+ways to do this, depending on the nature of the tag. Two kind of tags exists:
+
+- Simple tags are just text replacements. For instance a ``[b]Some Text[/b]`` should be replaced by
+``<b>Some Text</b>``
+
+- Complex tags require more than a simple regexp conversion. For instance the tag
+``[media=format]MediId[/media]`` requires to fetch the media in the database, to detect its type and
+then to to return the corresponding html tag.
+
+Add a new BBCode to the parser requires to add its definition. There are two ways of adding a
+definition.
+
+Create a simple BBCode tag via configuration
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+To create a simple BBCode definition, its possible to use a trivial configuration. The parameter
+``open_orchestra_bbcode.code_definitions`` contains an array of declared simple tags. Here is for
+instance the declaration of the bold tag:
+
+.. code-block:: yaml
+
+    bold:
+        tag: b
+        html: <strong>{param}</strong>
+
+- The key ``bold`` is the name of the definition used internally by the parser.
+
+- The ``html`` key is the html pattern of the code. The string ``{param}`` will be replaced by the
+content of the BBcode tag.
+
+- The ``tag`` key is the BBCode pattern of the tag. In this example the tag is ``[b]{param}[/b]``.
```

Comment:
The yaml is correct but I have to present keys not in the config order to easily introduce the string {param}. Presenting the keys in the config order would lead to a complex description of the string {param}

However I rewrite the sentence about "tag" which may be ambiguous. Let me know if it is ok for you when I have push it.